pkg/loki/loki.go

```diff
@@ -725,6 +725,12 @@ func (t *Loki) setupModuleManager() error {
 		}
 	}
 
+	// Add BloomStore dependency to IndexGateway in case bloom gateway component is enabled.
+	// This is needed because the blocks for the bloom gateway are resolved on the index gateway.
+	if t.Cfg.BloomGateway.Enabled {
+		deps[IndexGateway] = append(deps[IndexGateway], BloomStore)
```

Comment:
nit: I think it makes more sense to use this check in `initBloomStore` instead and just return a `nil` or `noop` impl if it's disabled. Reason being, module dependency is hard enough and it's nice to keep it conceptually simple + pass the implementation decisions downward to the implementor